cli/azd/pkg/auth/manager.go

```diff
@@ -432,14 +437,20 @@ func (m *Manager) LoginWithDeviceCode(
 		return nil, err
 	}
 
-	// Display the message to the end user as to what to do next, then block waiting for them to complete
-	// the flow.
-	fmt.Fprintln(deviceCodeWriter, code.Message())
+	console.Message(ctx, fmt.Sprintf("Start by copying the next code: %s", output.WithBold(code.UserCode())))
+	console.Message(ctx, "Then press enter and continue to log in from your browser...")
+	_ = bufio.NewScanner(console.Handles().Stdin).Scan()
```

Comment:
Could we add a `WaitForEnter` method to `console` that just waits for enter (so the implementation for AskerConsoler would look like this bufio Scanner stuff).

That might end up being a less leaky abstraction and may make it easier for folks who are mocking the console to handle this sort of stuff.

Reply:
I would really like to avoid adding more methods to Console interface (unless it is really necessary).
In this case, what we need is just a simple decoration for Message() and/or MessageUX() methods.  So, I have added a new console implementation called `paused_console` to decorate an existing console with the expected behavior of blocking until pressing enter.